x/subscription/keeper/subscription_test.go

```diff
@@ -1075,5 +1070,602 @@ func TestSubscriptionCuExhaustAndUpgrade(t *testing.T) {
+	require.Equal(t, consumerBalance, ts.GetBalance(consumerAcc.Addr))
+
+	ts.AdvanceBlock()
+
+	// Create plan with same index - higher price
+	mediumPlanExpensive := common.CreateMockPlan()
+	mediumPlanExpensive.Index = MEDIUM
+	mediumPlanExpensive.Price = common.NewCoin(ts.TokenDenom(), 300)
+	mediumPlanExpensive.Block = ts.BlockHeight()
+	ts.AddPlan(mediumPlanExpensive.Index, mediumPlanExpensive)
+
+	// Buy new plan
+	_, err = ts.TxSubscriptionBuy(consumerAddr, consumerAddr, mediumPlanCheaper.Index, int(startingDuration-1), false, true)
+	require.NotNil(t, err, "Same plan && cheaper && more duration -> Price: "+
+		mediumPlanCheaper.Price.Amount.MulRaw(startingDuration-1).String()) // 600
+
+	// Make sure the balance is not changed
+	require.Equal(t, consumerBalance, ts.GetBalance(consumerAcc.Addr))
+
+	// Buy new plan
+	_, err = ts.TxSubscriptionBuy(consumerAddr, consumerAddr, mediumPlanCheaper.Index, int(startingDuration-2), false, true)
+	require.NotNil(t, err, "Same plan && cheaper && more duration -> Price: "+
+		mediumPlanCheaper.Price.Amount.MulRaw(startingDuration-2).String()) // 500
+
+	// Make sure the balance is not changed
+	require.Equal(t, consumerBalance, ts.GetBalance(consumerAcc.Addr))
+}
+
+func TestSubscriptionAdvancePurchaseFailOnNoSubscription(t *testing.T) {
+	ts := newTester(t)
+	ts.SetupAccounts(1, 0, 0) // 1 sub, 0 adm, 0 dev
+
+	consumerAcc, consumerAddr := ts.Account("sub1")
+	premiumPlan := ts.Plan("premium")
+	consumerBalance := ts.GetBalance(consumerAcc.Addr)
+
+	// Advance purchase the subscription with no active subscription
+	newSubDuration := int64(4)
+	_, err := ts.TxSubscriptionBuy(consumerAddr, consumerAddr, premiumPlan.Index, int(newSubDuration), false, true)
+	require.NotNil(t, err)
+
+	// Verify that the consumer is not charged
+	require.Equal(t, consumerBalance, ts.GetBalance(consumerAcc.Addr))
+
+	// Verify that there is no new subscription
+	_, found := ts.getSubscription(consumerAddr)
+	require.False(t, found)
+}
+
+func TestSubscriptionAdvancePurchaseNewCreator(t *testing.T) {
+	ts := newTester(t)
+	ts.SetupAccounts(1, 0, 0) // 1 sub, 0 adm, 0 dev
+
+	creator1Acc, creator1Addr := ts.AddAccount("sugar1", 0, 20000)
+	creator1Balance := ts.GetBalance(creator1Acc.Addr)
+
+	creator2Acc, creator2Addr := ts.AddAccount("sugar2", 1, 20000)
+	creator2Balance := ts.GetBalance(creator2Acc.Addr)
+
+	consumerAcc, consumerAddr := ts.Account("sub1")
+	consumerBalance := ts.GetBalance(consumerAcc.Addr)
+	freePlan := ts.Plan("free")
+	premiumPlan := ts.Plan("premium")
+
+	// Buy free plan
+	freePlanDuration := int64(2)
+	_, err := ts.TxSubscriptionBuy(consumerAddr, consumerAddr, freePlan.Index, int(freePlanDuration), false, false)
+	require.Nil(t, err)
+	// Verify subscription found inside getSubscription
+	getSubscriptionAndFailTestIfNotFound(t, ts, consumerAddr)
+	consumerBalance -= freePlan.Price.Amount.MulRaw(freePlanDuration).Int64()
+	// Make sure that creator1 payed for the subscription
```

Comment:
paid

Reply:
Fixed in 4620db68491b3e8207000f6f58932aa066698814